lib/Access.php

```diff
@@ -236,38 +278,48 @@ public function readAttribute($dn, $attr, $filter = 'objectClass=*') {
 	 *
 	 * @param resource $cr the LDAP connection
 	 * @param string $dn
-	 * @param string $attribute
+	 * @param array $attributes (pass in 1 element, '', and this method will check the dn exists)
 	 * @param string $filter
 	 * @param int $maxResults
 	 * @return array|bool false if there was any error, true if an exists check
 	 *                    was performed and the requested DN found, array with the
 	 *                    returned data on a successful usual operation
 	 */
-	public function executeRead($cr, $dn, $attribute, $filter, $maxResults) {
-		$this->initPagedSearch($filter, array($dn), array($attribute), $maxResults, 0);
+	public function executeRead($cr, $dn, $attributes, $filter, $maxResults) {
+		$this->initPagedSearch($filter, array($dn), $attributes, $maxResults, 0);
 		$dn = $this->DNasBaseParameter($dn);
-		$rr = @$this->getLDAP()->read($cr, $dn, $filter, array($attribute));
+		$rr = @$this->getLDAP()->read($cr, $dn, $filter, $attributes);
 		if(!$this->getLDAP()->isResource($rr)) {
-			if ($attribute !== '') {
-				//do not throw this message on userExists check, irritates
-				Util::writeLog('user_ldap', 'readAttribute failed for DN '.$dn, Util::DEBUG);
-			}
+			Util::writeLog('user_ldap', 'readAttributes failed for DN '.$dn, Util::DEBUG);
 			//in case an error occurs , e.g. object does not exist
 			return false;
 		}
-		if ($attribute === '' && ($filter === 'objectclass=*' || $this->getLDAP()->countEntries($cr, $rr) === 1)) {
+
+		// If we were just checking the existence of a dn
+		if (
+			count($attributes) === 1
+			&& reset($attributes) === ''
+			&& ($filter === 'objectclass=*' || $this->getLDAP()->countEntries($cr, $rr) === 1)
+		) {
 			Util::writeLog('user_ldap', 'readAttribute: '.$dn.' found', Util::DEBUG);
 			return true;
 		}
 		$er = $this->getLDAP()->firstEntry($cr, $rr);
 		if(!$this->getLDAP()->isResource($er)) {
-			//did not match the filter, return false
+			// did not match the filter, return false
 			return false;
 		}
-		//LDAP attributes are not case sensitive
+		// LDAP attributes are not case sensitive
 		$result = Util::mb_array_change_key_case(
 			$this->getLDAP()->getAttributes($cr, $er), MB_CASE_LOWER, 'UTF-8');
 
+		// Hack - for some reason the DN is lost in the result - manually add it
+		$count = $result['count'];
+		unset($result['count']);
+		$result['dn'] = ['count' => 1, $dn];
+		$result[] = 'dn';
+		$result['count'] = $count++;
+
```

Comment:
@patrickjahns @butonic I don't know why, but ldap->getAttributes() just doesn't come back with the dn in the array so I had to hack to add it back on in the same format